en/views/helpers/form.rst

```diff
@@ -888,171 +1135,256 @@ put custom attributes on your radio buttons you can use an expanded format::
+
 Creating Select Pickers
------------------------
+~~~~~~~~~~~~~~~~~~~~~~~
 
 .. php:method:: select(string $fieldName, array $options, array $attributes)
 
-Creates a select element, populated with the items in ``$options``,
-with the option specified by ``$attributes['value']`` shown as selected by
-default. Set the 'empty' key in the ``$attributes`` variable to ``true`` (the default value is ``false``) to
-add a blank option with an empty value on the top of your dropdown list::
+* ``$fieldName`` - A field name in the form ``'modelname.fieldname'``. This
+  will provide the ``name`` attribute of the ``select`` element.
+* ``$options`` - An optional array containing the list of items for the select
+  picker. When this array is missing, the method will generate only the
+  empty ``select`` HTML element without any ``option`` elements inside it.
+* ``$attributes`` - An optional array including any of the
+  :ref:`general-input-options`, or of the :ref:`checkbox-radio-select-options`,
+  or of the select-specific attributes (see below), as well as any valid
+  HTML attributes.
 
-    $options = ['M' => 'Male', 'F' => 'Female'];
-    echo $this->Form->select('gender', $options, ['empty' => true]);
+Creates a ``select`` element, populated with the items from the ``$options``
+array. If ``$attributes['value']`` is provided, then the HTML ``option``
+element(s) which have the specified value(s) will be shown as selected when
+rendering the select picker.
 
-Will output:
+By default ``select`` uses the following widget templates::
+
+    'select' => '<select name="{{name}}"{{attrs}}>{{content}}</select>
+    'option' => '<option value="{{value}}"{{attrs}}>{{text}}</option>'
+
+May also use::
+
+    'optgroup' => '<optgroup label="{{label}}"{{attrs}}>{{content}}</optgroup>'
+    'selectMultiple' => '<select name="{{name}}[]" multiple="multiple"{{attrs}}>{{content}}</select>'
+
+**Attributes for Select Pickers**
+
+* ``'multiple'`` - If set to ``true`` allows multiple selections in the select
+  picker. If set to ``'checkbox'``, multiple checkboxes will be created instead.
+  Defaults to ``null``.
+
+* ``'escape'`` - Boolean. If ``true`` the contents of the ``option`` elements
+  inside the select picker will be HTML entity encoded. Defaults to ``true``.
+
+* ``'val'`` - Allows preselecting a value in the select picker.
+
+* ``'disabled'`` - Controls the ``disabled`` attribute. If set to ``true``
+  disables the whole select picker. If set to an array it will disable
+  only those specific ``option`` elements whose values are provided in
+  the array.
+
+The ``$options`` argument allows you to manually specify
+the contents of the ``option`` elements of a ``select`` input.
+
+E.g. ::
+
+  echo $this->Form->select('field', [1, 2, 3, 4, 5]);
+
+Output:
 
 .. code-block:: html
 
-    <select name="gender">
-    <option value=""></option>
-    <option value="M">Male</option>
-    <option value="F">Female</option>
-    </select>
+  <select name="field">
```

Comment:
Fixed in current commit. I've also went through the whole file and fixed the indent for a bunch of other code samples that were indented by only 3 or 2 spaces; now they should all be at 4 spaces for first indentation level, as you seem to have implicitly suggested. I've also tried to make sure that all extra indentation levels are 4 spaces each, as well.